src/kademlia/dht_storage.cpp

```diff
@@ -235,26 +232,20 @@ namespace
 					if (iter->addr.protocol() != protocol)
```

Comment:
Yeah, I realized this after I went to bed last night. Note to self: always sleep on a patch before submitting a PR.

I'm debating splitting the peers set into separate IPv4 and IPv6 sets so that the max_peers limit is applied to each one separately. It would be bad to not have any IPv6 peers because the set is full of IPv4 peers or vice-versa. I'm not sure that would ever be a problem in practice though.